src/useQueryParam.ts

```diff
@@ -31,11 +33,18 @@ export const useQueryParam = <D, D2 = D>(
 ): [D2 | undefined, (newValue: D, updateType?: UrlUpdateType) => void] => {
   const { history, location } = React.useContext(QueryParamContext);
 
-  // read in the raw query
   if (!rawQuery) {
-    rawQuery = React.useMemo(() => parseQueryString(location.search) || {}, [
-      location.search,
-    ]);
+    rawQuery = React.useMemo(() => {
+      let pathname = {}
+
+      if (typeof location === 'object' && typeof window !== undefined) {
+        pathname = parseQueryString(location.search)
+      } else if (typeof location === 'object') {
+        pathname = parseQueryURL(location.pathname).query
+      }
+
+      return parseQueryString(location.search) || {}
```

Comment:
I don't understand why you're configuring `pathname` and then not using it. Can you explain how this is supposed to work?

Reply:
Returning the `pathname` now. Thanks for the catch.